### ISSUE TITLE
Feature Request : Copy Post/Template, auto-suggest current date

### ISSUE THREAD

**Issue by moonwitch** [FauDroids/MrHyde#20]
- It would be handy to have the ability to copy a post or to copy a remade template. 
- When creating a new post/draft (new or copied), auto-fill in the current date fully highlighted. 

I am just listing these here after the mail I sent ;) 

I haven't a lot of Java experience, let alone compiling/testing Android apps. But I assume it would be possible to check for the existence of a "template" folder, and allow copy from there to posts/drafts. Need to think on this more before formulating an actual idea.

**Comment by Maddoc42** [comment]
Just split this issue in two. @moonwitch hopefully #21 captures your original intention

**Comment by Maddoc42** [comment]
Is there any specific reason for including the templates in the repository itself?

It might make sense to store the template information locally on the android device (e.g. shared preferences etc.) as this template information is relevant for MrHyde only. Or am I missing something?

**Comment by moonwitch** [comment]
@Maddoc42 Shared templates for MrHyde only would make sense, unless the user configured their Jekyll install to use a template already -> https://github.com/moonwitch/noaggro 
The Rakefile I use already uses a crude template for new posts. 

```ruby
namespace :create do # create a namespace so I don't have to duplicate as much
  desc "Create a post in _posts" # rake post["title","cat","date"]
  task :post, [:title, :category, :dates] do |t, args|
    check_title(args.title) # Title is required as field
    args.with_defaults(dates: Time.now.strftime("%Y-%m-%d")) # adding default
    args.with_defaults(category: "blog") # adding a default value

    # Set the filename to date-title.md
    filename = args.dates + "-" + transform_to_slug(args.title) + ".md"

    # Create the file if it doesn't already exist
    if File.exists?(POSTS + filename) then
      # Open existing file in text editor
      sh "atom \"#{POSTS}#{filename}\""
    else
      # Create post, based on the parameters passed to the rake task
      File.open(POSTS + filename, "w") do |f|
        f.puts "---"
        f.puts "layout: post"
        f.puts "image: optional"
        f.puts "title: #{args.title}"
        f.puts "date: #{args.dates}"
        f.puts "category: #{args.category}"
        f.puts "comments: true"
        f.puts "tags: "
        f.puts "permalink: "
        f.puts "---"

        # Open the new file in text editor
        sh "atom \"#{POSTS}#{filename}\""
      end # file
    end # if
  end # task
end # namespace create
```

I thought something along those lines might work, but figured a copy from a default post/draft would be easier to implement. Say you have 2 possible templates depending on where you choose the Copy/New action (posts or drafts directory):
```
---
layout: default
title: {{post.title}}
date: {{todays date}}
categories: {{if supplied}}
tags: {{if supplied}}
excerpt: ""
---
```

**Comment by Maddoc42** [comment]
The current default template for posts / drafts is:
```
---
layout: post
title: <your title>
---
```

which is populated with the proper title when creating a new entry. From there we could allow users to modify / extends this template.

I'm not so sure about the copying from existing posts. Which one would you choose? What if the user decides to extend his "default template" but does not update old posts?

On the other hand, copying the front matter from a random posts during the first start of MrHyde and use that as a "default" starting template would be awesome. This is then basically a "smart" default template which is created in the beginning and can be modified later.

**Comment by moonwitch** [comment]
I completely agree with the "copy the front matter from a post", that was pretty much why I wanted to add a copy - so the custom front matter can be reused... :) Awesome.
